File: week-7/protoseo/두 큐 합 같게 만들기.java
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        long sum = 0;
        long now = 0;
        int l = 0;
        int r = queue1.length - 1;
        int n = queue1.length + queue2.length;
        int[] queue = new int[n];

        for (int i = 0; i < queue1.length; i++) {
            sum += queue1[i];
            queue[i] = queue1[i];
        }
        now = sum;
        for (int i = 0; i < queue2.length; i++) {
            sum += queue2[i];
            queue[i + queue1.length] = queue2[i];
        }

        boolean isAnswer = false;
        int answer = 0;
        while (l <= r && r < n - 1) {
            if (now < (sum / 2)) {
                now += queue[++r];
                answer++;
            } else if (now > (sum / 2)) {
                now -= queue[l++];
                answer++;
            }
            if (now == (sum / 2)) {
                isAnswer = true;
                break;
            }
        }
        if (isAnswer) {
            return answer;
        }
        return -1;
    }
}

File: week-7/dahyen0o/두-큐-합-같게-만들기.py
def solution(queue1, queue2):
    answer = 0
    sum_q1 = sum(queue1)
    sum_q2 = sum(queue2)
    mean_sum = (sum_q1 + sum_q2) / 2
    if mean_sum != int(mean_sum):
        return -1
    
    i = 0
    j = 0
    total_len = len(queue1) * 2
    # while i < len(queue1) and j < len(queue2) and sum_q1 != sum_q2:
    while i < total_len and j < total_len and sum_q1 != sum_q2:
        if sum_q1 > mean_sum:
            sum_q1 -= queue1[i]
            sum_q2 += queue1[i]
            queue2.append(queue1[i])
            i += 1

        else:# sum_q2 > mean_sum:
            sum_q2 -= queue2[j]
            sum_q1 += queue2[j]
            queue1.append(queue2[j])
            j += 1
    
    return -1 if sum_q1 != sum_q2 else i + j
